run.go

```diff
@@ -201,37 +211,65 @@ func runff(ff *funcfile, stdin io.Reader, stdout, stderr io.Writer, method strin
 		// reqID := id.New().String()
 		// I'm starting to think maybe `fn run` locally should work the same whether sync or async?  Or how would we allow to test the output?
 	}
-	body := "" // used for hot functions
-	if format == HttpFormat {
-		// let's swap out stdin for http formatted message
-		input := []byte("")
-		if stdin != nil {
-			input, err = ioutil.ReadAll(stdin)
-			if err != nil {
-				return fmt.Errorf("error reading from stdin: %v", err)
-			}
-		}
 
+	input := []byte("")
+	if stdin != nil {
+		input, err = ioutil.ReadAll(stdin)
+		if err != nil {
+			return fmt.Errorf("error reading from STDIN: %v", err)
+		}
+	}
+	h := http.Header{}
+	h.Set("FN_CALL_ID", "12345678901234567890123456")
+	h.Set("FN_METHOD", method)
+	h.Set("FN_REQUEST_URL", LocalTestURL)
+	h.Set("FN_APP_NAME", "myapp")
+	h.Set("FN_PATH", ff.Path)
+	h.Set("FN_FORMAT", ff.Format)
+	h.Set("FN_MEMORY", strconv.Itoa(int(ff.Memory)))
+	h.Set("FN_TYPE", "sync")
+	h.Set("FN_DEADLINE", strfmt.DateTime(
```

Comment:
this should use the real ff.Timeout if defined.

Reply:
indeed, if not defined we should use default value which is 30s.